backend/corpora/common/utils/db_utils.py

```diff
@@ -84,3 +85,17 @@ def wrapper_decorator(*args, **kwargs):
             db.close()
 
     return wrapper_decorator
+
+
+@contextmanager
+def db_session_manager():
+    try:
+        db = DbUtils()
+        yield db
+    except SQLAlchemyError:
+        db.session.rollback()
```

Comment:
Can you add a test case for errors/rollback?

Reply:
yes I can